test/cloudtest/test_compute.py

```diff
@@ -126,16 +126,9 @@ def get_node_selector(
 def test_disk_label(mz: MaterializeApplication) -> None:
     """Test that cluster replicas have the correct materialize.cloud/disk labels"""
 
-    # If cluster_always_use_disk is set to true, it will take precedence over the DISK keyword in CREATE CLUSTER.
-    mz.environmentd.sql(
-        "ALTER SYSTEM SET cluster_always_use_disk = false;",
-        port="internal",
-        user="mz_system",
-    )
-
     for value in ("true", "false"):
         mz.environmentd.sql(
-            f"CREATE CLUSTER disk_{value} MANAGED, SIZE = 'scale=1,workers=2,nodisk', DISK = {value}"
+            f"CREATE CLUSTER disk_{value} MANAGED, SIZE = 'scale=1,workers=2,legacy', DISK = {value}"
```

Comment:
How is this test passing? Isn't the `else` branch of the below `if value == "true":` intended to check whether we indeed don't have a disk if `disk` is set to false? But with the PR, we should be always getting a disk.